src/main/java/com/capitalone/dashboard/service/AutoDiscoveryServiceImpl.java

```diff
@@ -49,4 +56,38 @@ private AutoDiscovery requestToAutoiscovery(AutoDiscoveryRemoteRequest request)
                 request.getDeploymentEntries(), request.getLibraryScanEntries(), request.getFunctionalTestEntries(), request.getArtifactEntries(),
                 request.getStaticCodeEntries(), request.getFeatureEntries());
     }
+
+
+    /**
+     * Update the AutoDiscovery Entries' status from the request.
+     * @param autoDiscovery
+     * @param request
+     */
+    private void updateAutoiscovery(AutoDiscovery autoDiscovery, AutoDiscoveryRemoteRequest request) {
+        updateEntryStatus(request.getCodeRepoEntries(), autoDiscovery.getCodeRepoEntries());
+        updateEntryStatus(request.getBuildEntries(), autoDiscovery.getBuildEntries());
+        updateEntryStatus(request.getSecurityScanEntries(), autoDiscovery.getSecurityScanEntries());
+        updateEntryStatus(request.getDeploymentEntries(), autoDiscovery.getDeploymentEntries());
+        updateEntryStatus(request.getLibraryScanEntries(), autoDiscovery.getLibraryScanEntries());
+        updateEntryStatus(request.getFunctionalTestEntries(), autoDiscovery.getFunctionalTestEntries());
+        updateEntryStatus(request.getArtifactEntries(), autoDiscovery.getArtifactEntries());
+        updateEntryStatus(request.getStaticCodeEntries(), autoDiscovery.getStaticCodeEntries());
+        updateEntryStatus(request.getFeatureEntries(), autoDiscovery.getFeatureEntries());
+    }
+
+    /**
+     * Update the AutoDiscovery Entries' status from source to target.
+     * @param source
+     * @param target
+     */
+    private void updateEntryStatus(List<AutoDiscoveredEntry> source, List<AutoDiscoveredEntry> target) {
+        for (AutoDiscoveredEntry srcEntry : source) {
+            target.stream().forEach(entry -> {
+                if(entry.getOptions().equals(srcEntry.getOptions())){
```

Comment:
@danielyhuang  does this checks references or values

Reply:
Yes, it compares key-value pair if both are String type, which is our case. If the value is a List, then we need special handling.